configfs.c

```diff
@@ -51,25 +52,85 @@ int tcmu_get_cfgfs_int(const char *path)
 	return val;
 }
 
-int tcmu_get_attribute(struct tcmu_device *dev, const char *name)
+int tcmu_cfgfs_dev_get_attr_int(struct tcmu_device *dev, const char *name)
 {
 	char path[PATH_MAX];
 
 	snprintf(path, sizeof(path), CFGFS_CORE"/%s/%s/attrib/%s",
 		 dev->tcm_hba_name, dev->tcm_dev_name, name);
-	return tcmu_get_cfgfs_int(path);
+	return tcmu_cfgfs_get_int(path);
 }
 
-int tcmu_set_control(struct tcmu_device *dev, const char *key, unsigned long val)
+uint64_t tcmu_cfgfs_dev_get_info_u64(struct tcmu_device *dev, const char *name,
+				     int *fn_ret)
+{
+	int fd;
+	char path[PATH_MAX];
+	char buf[CFGFS_BUF_SIZE];
+	ssize_t ret;
+	char *rover;
+	char *search_pattern;
+	uint64_t val;
+
+	*fn_ret = 0;
+	snprintf(path, sizeof(path), CFGFS_CORE"/%s/%s/info",
+		 dev->tcm_hba_name, dev->tcm_dev_name);
+
+	fd = open(path, O_RDONLY);
+	if (fd == -1) {
+		tcmu_err("Could not open configfs to read dev info: %s\n",
```

Comment:
Here we can use the tcmu_dev_XXX helper.

Reply:
Oh yeah, that was my fault. I gave baihuahua a bad review comment. I forgot that was my other TODO item. I want to fix how the log and config code is setup before it this is released and then add the log functions to the common lib header or install the libtcmu_log.h so it can be used in the lib.

Today where tcmu-runner has to destroy those objects in the odd order because of the dependency seems too easy to mess up for lib users, so I just wanted to fix that up.

Reply:
I agree with @mikechristie here.

Reply:
It looks like @baihuahua is going to tackle these issues. I am going to merge what we have now so it is easier to work on it.

Reply:
Great. Thanks a lot.